src/mixins/itext_behavior.mixin.js

```diff
@@ -152,24 +160,23 @@
     },
 
     /**
-     * Aborts cursor animation and clears all timeouts
+     * Aborts cursor animation, clears all timeouts and clear textarea context if necessary
      */
     abortCursorAnimation: function() {
-      var shouldClear = this._currentTickState || this._currentTickCompleteState,
-          canvas = this.canvas;
+      var shouldClear = this._currentTickState || this._currentTickCompleteState;
       this._currentTickState && this._currentTickState.abort();
       this._currentTickCompleteState && this._currentTickCompleteState.abort();
 
       clearTimeout(this._cursorTimeout1);
       clearTimeout(this._cursorTimeout2);
 
       this._currentCursorOpacity = 0;
-      // to clear just itext area we need to transform the context
-      // it may not be worth it
-      if (shouldClear && canvas) {
-        canvas.clearContext(canvas.contextTop || canvas.contextContainer);
-      }
 
+      //  make sure we clear context even if instance is not editing
+      if (shouldClear) {
+        var ctx = this._clearContextTop();
+        ctx && ctx.restore();
```

Comment:
why this lonely restore? where the save counterpar is?

Reply:
`clearContextTop` -> `_clearContextTop` -> `prepareContextTop` -> `ctx.save()`

Reply:
yes i found it later, at some point we need to make it more evident. Lonely restore or save are bugs ready to happen

Reply:
agreed.
I think we can do something like
```js
runInContext(callback: (ctx: Canvas2DRenderingContext) => any)
```
I am more than willing to migrate to that right away.

Reply:
brush suffers from this as well BTW. There's an open issue related.

Reply:
contextTop i think is the only place where we do this in a non straight forward way.
If there is more let's find a strategy.
Can you link me the issue?

Reply:
#7732